src/librustc_mir/const_eval/mod.rs

```diff
@@ -43,6 +45,12 @@ pub(crate) fn destructure_const<'tcx>(
 
     let variant = ecx.read_discriminant(op).unwrap().1;
 
+    if let ty::Adt(def, _) = val.ty.kind {
+        if def.variants.is_empty() {
```

Comment:
You could merge this into the match below. Imo it's ok to have an arm that does an early return when all the others yield a value.

Reply:
Fixed!